tests/rptest/tests/cluster_quota_test.py

```diff
@@ -100,3 +105,147 @@ def test_partition_throttle_mechanism(self):
         assert foo_response[0]['ErrorCode'] == 0  # success
         assert baz_response[0]['ErrorCode'] == 89  # throttling_quota_exceeded
         assert response['ThrottleMillis'] > 0
+
+
+class ClusterRateQuotaTest(RedpandaTest):
+    """
+    Ducktape tests for rate quota
+    """
+    topics = (TopicSpec(), )
+
+    def __init__(self, *args, **kwargs):
+        target_quota_byte_rate = 1048576
+        self.message_size = 1024
+        self.break_quota_message_amount = int(target_quota_byte_rate /
+                                              self.message_size * 10)
+        self.max_partition_fetch_bytes = self.break_quota_message_amount * self.message_size
+        additional_options = {
+            "max_kafka_throttle_delay_ms": 10,
+            "target_quota_byte_rate": target_quota_byte_rate,
+            "target_fetch_quota_byte_rate": target_quota_byte_rate,
+        }
+        super().__init__(*args,
+                         num_brokers=3,
+                         extra_rp_conf=additional_options,
+                         resource_settings=ResourceSettings(num_cpus=1),
+                         **kwargs)
+        self.rpk = RpkTool(self.redpanda)
+
+    def init_test_data(self):
+        wait_until(lambda: len(list(self.rpk.describe_topic(self.topic))) != 0,
+                   10)
+        self.leader_node = self.redpanda.broker_address(
+            self.redpanda.get_node(
+                next(self.rpk.describe_topic(self.topic)).leader))
+
+        self.msg = "".join(
+            random.choice(string.ascii_lowercase)
+            for _ in range(self.message_size))
+
+    def produce(self, producer, amount):
+        response_futures = [
+            producer.send(self.topic, self.msg) for _ in range(amount)
+        ]
+        for f in response_futures:
+            f.get(1)
+
+    @cluster(num_nodes=3)
+    def test_client_response_throttle_mechanism(self):
+        """
+        Ensure response size rate throttle works
+        """
+        self.init_test_data()
+
+        producer = KafkaProducer(acks="all",
+                                 bootstrap_servers=self.leader_node,
+                                 value_serializer=str.encode,
+                                 retries=1,
+                                 client_id="producer")
+
+        consumer = KafkaConsumer(
+            self.topic,
+            bootstrap_servers=self.leader_node,
+            client_id="consumer",
+            consumer_timeout_ms=1000,
+            max_partition_fetch_bytes=self.max_partition_fetch_bytes,
+            auto_offset_reset='earliest',
+            enable_auto_commit=False)
+
+        self.produce(producer, self.break_quota_message_amount * 2)
+
+        # Consume under the quota
+        consumer.poll(timeout_ms=1000, max_records=10)
+        assert consumer.metrics(
+        )["consumer-fetch-manager-metrics"]["fetch-throttle-time-max"] == 0
```

Comment:
(discuss) I could not see the `fetch-throttle-time-max` metric in this PR... have I missed something?

Reply:
This metric is calculated by KafkaConsumer client from `throttle_time_ms` field of every fetch response.